runtime/parachains/src/ump.rs

```diff
@@ -337,12 +336,21 @@ impl<T: Config> Module<T> {
 				// if so - bail.
 				break;
 			}
+			let max_weight = if used_weight_so_far == 0 {
+				// we increase the amount of weight that we're allowed to use on the first message to try to prevent
+				// the possibility of blockage of the queue.
```

Comment:
this scenario is not quite clear to me yet

Reply:
The original UMP dispatcher guaraunteed one dispatch per block minimum with the weight limitation being a soft cap. This kept things running the case of a single heavy message. Now, we don't quite guarantee that, but the soft-cap gets lifted to this hard-cap for the first message per block to try to mitigate any blockage.